lib/src/rewrite.rs

```diff
@@ -471,85 +472,91 @@ pub struct MoveCommitsStats {
     pub num_abandoned: u32,
 }
 
+pub enum MoveCommitsTarget {
+    /// The commits to be moved. Commits should be mutable and in reverse
+    /// topological order.
+    Commits(Vec<Commit>),
+    /// The root commits to be moved, along with all their descendants.
+    /// Root commits should be mutable and in reverse topological order.
```

Comment:
nit: I don't think roots have to be in topological order. CLI wouldn't guarantee that.